pex/resolve/lockfile/updater.py

```diff
@@ -316,11 +316,40 @@ def update_resolve(
                 self.update_constraints_by_project_name
                 and project_name not in self.update_constraints_by_project_name
             ):
-                assert updated_pin == original_pin
-                assert updated_requirement.artifact == locked_requirement.artifact
+                assert updated_pin == original_pin, (
+                    "The locked requirement {original} should have been undisturbed by the lock "
+                    "update, but it changed to {updated}.".format(
+                        original=original_pin.as_requirement(), updated=updated_pin.as_requirement()
+                    )
+                )
+                assert updated_requirement.artifact == locked_requirement.artifact, (
```

Comment:
@cburroughs, although these asserts should never happen and thus not need good error messages, this one did trip which made your life harder debugging; so ... perhaps if there is a regression the new IT does not catch this helps some day.